dartagnan/src/test/java/com/dat3m/dartagnan/inlineAsm/armv8/ck/AsmCkArmv8Test.java

```diff
@@ -56,7 +56,7 @@ public static Iterable<Object[]> data() throws IOException {
             {"mcslock", 2, Result.PASS},
             {"ticketlock", 1, Result.PASS},
             {"spsc_queue", 1, Result.PASS},
-            // {"stack_empty", 1, Result.PASS}, // https://github.com/hernanponcedeleon/Dat3M/issues/817
+            {"stack_empty", 2, Result.UNKNOWN},
```

Comment:
Is this version a lot slower than the `XYZ_assume` one? IIRC, the assume version returned PASS in like 30 seconds with B=3 using Z3, so I'm wondering why you keep the bound at 2.

Reply:
The assume version was not checking the case where the call to pop fails. Using a while (as one should) make it much slower. It took 28 min to get UNKNOWN with B=3 on my machine. 

Due to the side-effectful spinloop (causing the problem fixed in #824), isn't the case that the new liveness theory would never return PASS?

Reply:
The counterexample found before fixing #824 was not globally side-effectful: the non-terminating loop failed all of its conditional stores, so it had only local side-effects by updating some status register. We can detect repetition of local states.
I also started a run locally and I see that it does indeed take a lot longer, so the assume statement was quite powerful.